src/basic/random-util.c

```diff
@@ -123,19 +123,44 @@ void initialize_srand(void) {
         srand_called = true;
 }
 
-void random_bytes(void *p, size_t n) {
+#if RAND_MAX >= INT_MAX
+#  define RAND_STEP 4
+#elif RAND_MAX >= SHORT_INT_MAX
+#  define RAND_STEP 2
+#else
+#  define RAND_STEP 1
+#endif
+
+void pseudorandom_bytes(void *p, size_t n) {
         uint8_t *q;
+
+        initialize_srand();
+
+        for (q = p; q < (uint8_t*) p + n; q += RAND_STEP) {
+                int rr = rand();
```

Comment:
bit shifting on signed integers is always weird. Maybe turn this into an "unsigned" immediately? rand() only returns positive anyway, and then everything makes a ton more sense, to me, at least.

(btw, CODING_STYLE suggests not mixing variable declarations and funciton invocations in one line)